### PR TITLE
docs: add contributing including release process

### PR DESCRIPTION
**Pull request by aneojgurhem**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+# Contributing Guide
+
+Hi! We are very happy that you are interested in contributing to ArmoniK.Api. Before submitting your contribution, please make sure to take a moment and read through the following guide.
```
Comment (ngruelaneo) [inline CONTRIBUTING.md:3]:
```suggestion
Hi! We are very happy that you are interested in contributing to ArmoniK.Core project. Before submitting your contribution, please make sure to take a moment and read through the following guide.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+## Release Process
+
+When necessary, maintainers can release a new version. This new version will publish packages to registries.
+
+### Release a new version
+
+> Replace <version> with the new version number
+
+1. Install .Net latest release version (https://dotnet.microsoft.com/en-us/download)
```
Comment (ngruelaneo) [inline CONTRIBUTING.md:23]: not sure that we should always target the latest release, eventually the latest stable.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+2. Make sure that ArmoniK.Api.* versions are releases
```
Comment (ngruelaneo) [inline CONTRIBUTING.md:24]:
```suggestion
2. Make sure that ArmoniK.Core.* versions are releases
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  1. `git grep '"ArmoniK\.Api\.'` to check which version is used. You can also see that with any editor in csproj files.
```
Comment (ngruelaneo) [inline CONTRIBUTING.md:25]:
```suggestion
  1. `git grep '"ArmoniK\.Core\.'` to check which version is used. You can also see that with any editor in csproj files.
```

File: CONTRIBUTING.md (hunk 3 above)
Comment (aneojgurhem) [reply CONTRIBUTING.md:24]: core depends on releases from API

File: CONTRIBUTING.md (hunk 4 above)
Comment (aneojgurhem) [reply CONTRIBUTING.md:25]: that does not make any sense here

File: CONTRIBUTING.md (hunk 2 above)
Comment (aneojgurhem) [reply CONTRIBUTING.md:23]: that is for the compiler. The target is defined by the project and you cannot change it without modifying code

File: CONTRIBUTING.md (hunk 4 above)
Comment (ngruelaneo) [reply CONTRIBUTING.md:21]: my bad

**Review by ngruelaneo — APPROVED** [review]

**Pull request merged**